Process .meta file and add connection_spec fields.

gfr10598 opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/m-lab/etl/80)
<!-- Reviewable:end -->

gfr10598 commented:
Reviewed 1 of 5 files at r2.
Review status: 0 of 7 files reviewed at latest revision, 1 unresolved discussion.

---

*[parser/ndt.go, line 36 at r3](https://reviewable.io:443/reviews/m-lab/etl/80#-KkSfVGdZYwbLmuRodmi:-KkSfVGdZYwbLmuRodmj:bgv8kme) ([raw file](https://github.com/m-lab/etl/blob/3590ed01e41a4c6e8d3994d8fafe45f56abfbb36/parser/ndt.go#L36)):*
> ```Go
> const dateDir = `^(?P<dir>\d{4}/\d{2}/\d{2}/)?`
> 
> // TODO(NOW) - use time.Parse to parse this part of the filename.
> ```

This "NOW" is a mistake on my part.  It is intended for the IP parsing, which is a comment that I lost on merge at some point.
This one is not at all urgent.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/etl/80)*
<!-- Sent from Reviewable.io -->

stephen-soltesz commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

The ndt parsing is getting complex (necessarily). Are tests for the various meta file cases planned?

---

Reviewed 1 of 4 files at r3.
Review status: 1 of 7 files reviewed at latest revision, 8 unresolved discussions.

---

*[metrics/metrics.go, line 101 at r3](https://reviewable.io:443/reviews/m-lab/etl/80#-KkSJzofFWytlrnI45zO:-KkSJzofFWytlrnI45zP:by9kct0) ([raw file](https://github.com/m-lab/etl/blob/3590ed01e41a4c6e8d3994d8fafe45f56abfbb36/metrics/metrics.go#L101)):*
> ```Go
> 	//
> 	// Provides metrics:
> 	//   etl_funny_tests{type}
> ```

nit: the example is missing some labels.

---

*[parser/ndt.go, line 99 at r3](https://reviewable.io:443/reviews/m-lab/etl/80#-KkSKXznbvw9j1IOaeRm:-KkSKXznbvw9j1IOaeRn:b-q336ic) ([raw file](https://github.com/m-lab/etl/blob/3590ed01e41a4c6e8d3994d8fafe45f56abfbb36/parser/ndt.go#L99)):*
> ```Go
> 	timestamp string // The unique timestamp common across all files in current batch.
> 
> 	// TODO(dev) Sometimes NDT writes multiple copies of c2s or s2c.  We need to save them
> ```

wat? how is this possible?

---

*[parser/ndt.go, line 268 at r3](https://reviewable.io:443/reviews/m-lab/etl/80#-KkSpSDfGH1S_jz1jpUn:-KkSpSDfGH1S_jz1jpUo:bydybjn) ([raw file](https://github.com/m-lab/etl/blob/3590ed01e41a4c6e8d3994d8fafe45f56abfbb36/parser/ndt.go#L268)):*
> ```Go
> }
> 
> // parseMetaFile converts the raw content into key value map.
> ```

Please add a short summary of the expected format of a meta file.

---

*[parser/ndt.go, line 286 at r3](https://reviewable.io:443/reviews/m-lab/etl/80#-KkSplx0jfa_Qbj2IHfE:-KkSplx0jfa_Qbj2IHfF:b-xjjh7x) ([raw file](https://github.com/m-lab/etl/blob/3590ed01e41a4c6e8d3994d8fafe45f56abfbb36/parser/ndt.go#L286)):*
> ```Go
> 			continue
> 		}
> 		result[kv[0]] = kv[1]
> ```

I'm 100% sure that some of the archival data is corrupted with binary data. Since we expect the keys and values to be printable ascii, we might plan to unconditionally filter all non-ascii characters.

---

*[schema/ndt.json, line 1 at r4](https://reviewable.io:443/reviews/m-lab/etl/80#-KkSvoHsaiqjt4l5vY95:-KkStMv-thKjnYVDT_Sb:b-nfc3yk) ([raw file](https://github.com/m-lab/etl/blob/d9875d19ac8fd911a19adfc36a74753231162e19/schema/ndt.json#L1)):*
> ```JSON
> [
> ```

When will this file be used?

---

*[schema/web100.go, line 19 at r3](https://reviewable.io:443/reviews/m-lab/etl/80#-KkStlcgEkEX6FQrz4Sd:-KkStlcgEkEX6FQrz4Se:b-3xuckk) ([raw file](https://github.com/m-lab/etl/blob/3590ed01e41a4c6e8d3994d8fafe45f56abfbb36/schema/web100.go#L19)):*
> ```Go
> }
> 
> // SetInt64 saves an int64 in a field with the given name.
> ```

nit: The docstring is out of sync.

This is for the latitude / longitude values? I don't see it used anywhere yet.

---

*[task/task.go, line 96 at r3](https://reviewable.io:443/reviews/m-lab/etl/80#-KkSvGpQjjzD9XEDcr-Z:-KkSvGpRcSPCkCLKjqzF:b-bgj1bd) ([raw file](https://github.com/m-lab/etl/blob/3590ed01e41a4c6e8d3994d8fafe45f56abfbb36/task/task.go#L96)):*
> ```Go
> 	}
> 	// TODO - make this debug or remove
> 	log.Printf("%s: %d files, %d nil data, %d rows", tt.FullTableName(), files, nilData, tt.Count())
> ```

Also reporting the `taskFileName` here could be helpful.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/etl/80#-:-KkSvtqYVrFKTjruLHUR:b40crkc)*
<!-- Sent from Reviewable.io -->

gfr10598 commented:
Review status: 1 of 9 files reviewed at latest revision, 8 unresolved discussions.

---

*[metrics/metrics.go, line 101 at r3](https://reviewable.io:443/reviews/m-lab/etl/80#-KkSJzofFWytlrnI45zO:-KkWHiy7ZbZUNvgv5lsl:b-896fix) ([raw file](https://github.com/m-lab/etl/blob/3590ed01e41a4c6e8d3994d8fafe45f56abfbb36/metrics/metrics.go#L101)):*
<details><summary><i>Previously, stephen-soltesz (Stephen Soltesz) wrote…</i></summary><blockquote>

nit: the example is missing some labels.
</blockquote></details>

Done.

---

*[parser/ndt.go, line 99 at r3](https://reviewable.io:443/reviews/m-lab/etl/80#-KkSKXznbvw9j1IOaeRm:-KkWHlQdrAP2WEkVrLx7:b5m8idh) ([raw file](https://github.com/m-lab/etl/blob/3590ed01e41a4c6e8d3994d8fafe45f56abfbb36/parser/ndt.go#L99)):*
<details><summary><i>Previously, stephen-soltesz (Stephen Soltesz) wrote…</i></summary><blockquote>

wat? how is this possible?
</blockquote></details>

Heh.  Peter confirmed it in the NDT code.  Errors may cause test to be retried, but the old file is left behind, though it seems weird that the name is different.

---

*[parser/ndt.go, line 268 at r3](https://reviewable.io:443/reviews/m-lab/etl/80#-KkSpSDfGH1S_jz1jpUn:-KkWJYzPmXib1w8L5H9F:b-896fix) ([raw file](https://github.com/m-lab/etl/blob/3590ed01e41a4c6e8d3994d8fafe45f56abfbb36/parser/ndt.go#L268)):*
<details><summary><i>Previously, stephen-soltesz (Stephen Soltesz) wrote…</i></summary><blockquote>

Please add a short summary of the expected format of a meta file.
</blockquote></details>

Done.

---

*[parser/ndt.go, line 286 at r3](https://reviewable.io:443/reviews/m-lab/etl/80#-KkSplx0jfa_Qbj2IHfE:-KkWJvpTjy6G8b6CyQzh:b-hukn99) ([raw file](https://github.com/m-lab/etl/blob/3590ed01e41a4c6e8d3994d8fafe45f56abfbb36/parser/ndt.go#L286)):*
<details><summary><i>Previously, stephen-soltesz (Stephen Soltesz) wrote…</i></summary><blockquote>

I'm 100% sure that some of the archival data is corrupted with binary data. Since we expect the keys and values to be printable ascii, we might plan to unconditionally filter all non-ascii characters.
</blockquote></details>

Added TODO.

---

*[schema/ndt.json, line 1 at r4](https://reviewable.io:443/reviews/m-lab/etl/80#-KkSvoHsaiqjt4l5vY95:-KkWK66fyEk3cEy6BKwT:b-tegtxe) ([raw file](https://github.com/m-lab/etl/blob/d9875d19ac8fd911a19adfc36a74753231162e19/schema/ndt.json#L1)):*
<details><summary><i>Previously, stephen-soltesz (Stephen Soltesz) wrote…</i></summary><blockquote>

When will this file be used?
</blockquote></details>

It should be used for reference, and for creating new table instances.
Unfortunately, json doesn't support comments.  8-(
Added a README.md

---

*[schema/web100.go, line 19 at r3](https://reviewable.io:443/reviews/m-lab/etl/80#-KkStlcgEkEX6FQrz4Sd:-KkWN6JFgrCl6NeNfcUF:bmi5dlp) ([raw file](https://github.com/m-lab/etl/blob/3590ed01e41a4c6e8d3994d8fafe45f56abfbb36/schema/web100.go#L19)):*
<details><summary><i>Previously, stephen-soltesz (Stephen Soltesz) wrote…</i></summary><blockquote>

nit: The docstring is out of sync.

This is for the latitude / longitude values? I don't see it used anywhere yet.
</blockquote></details>

Removing until needed.

---

*[task/task.go, line 96 at r3](https://reviewable.io:443/reviews/m-lab/etl/80#-KkSvGpQjjzD9XEDcr-Z:-KkWNkJUzYguoh_BwHow:b-896fix) ([raw file](https://github.com/m-lab/etl/blob/3590ed01e41a4c6e8d3994d8fafe45f56abfbb36/task/task.go#L96)):*
<details><summary><i>Previously, stephen-soltesz (Stephen Soltesz) wrote…</i></summary><blockquote>

Also reporting the `taskFileName` here could be helpful.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/etl/80)*
<!-- Sent from Reviewable.io -->
